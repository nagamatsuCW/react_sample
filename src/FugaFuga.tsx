import * as React from 'react'
import { connect } from "react-redux";
import { increment } from './action'
import { StoreState } from './reducer'
import { Dispatch } from 'redux'

const FugaFuga:React.FC<Props> = (props) => {
  return(
    <div id="test">
      FugaaaPiyoooooo{props.count}
      <button onClick={props.increment}>increment</button>
    </div>
  )
}

function mapStateToProps(state:StoreState) {
  return {
      count: state.counter.count
  };
}

function mapDispatchToProps(dispatch: Dispatch) {
  return {
    increment: () => { dispatch(increment()) }
  }
}

// type Props = ownProps & ReturnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>;
type Props = ReturnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FugaFuga);
