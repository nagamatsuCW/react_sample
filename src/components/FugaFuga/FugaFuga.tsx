import * as React from 'react'
import { Props } from './index'

const FugaFuga:React.FC<Props> = (props) => {
  return(
    <div id="test">
      FugaaaPiyoooooo
      <div id="count">
        {props.count}
      </div>
      <button id="incrementButton" onClick={props.increment}>increment</button>
    </div>
  )
}

export default FugaFuga