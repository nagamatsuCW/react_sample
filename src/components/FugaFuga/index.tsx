import { connect } from "react-redux";
import { increment, nameChange } from '../../action'
import { StoreState } from '../../reducer'
import { Dispatch } from 'redux'
import FugaFuga from './FugaFuga'


function mapStateToProps(state:StoreState) {
  return {
      count: state.counter.count,
      name: state.counter.name
  };
}

function mapDispatchToProps(dispatch: Dispatch) {
  return {
    increment: () => { dispatch(increment()) },
    nameChange: (value: string) => { dispatch(nameChange({name: value})) }
  }
}

// type Props = ownProps & RetuZrnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>;
export type Props = ReturnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FugaFuga);
