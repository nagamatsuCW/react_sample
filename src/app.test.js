import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import FugaFuga from './components/FugaFuga/FugaFuga';
Enzyme.configure({ adapter: new Adapter() });

function sum(a, b) {
  return a + b;
}
module.exports = sum;


test('adds 1 + 2 to equal 3', () => {
  expect(sum(1, 2)).toBe(3);
});

describe('FugaFuga.tsx', () => {

  const makeComponent = (args = {}) => (<FugaFuga  />);

  test('show', () => {
    const component = shallow(makeComponent());
    expect(component.find('#test').text()).toEqual('FugaaaPiyooooooincrement');
  })

})