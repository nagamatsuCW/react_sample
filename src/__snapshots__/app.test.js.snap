// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component 1`] = `
<div>
  FugaaaaaaIndexaaaaaa
</div>
`;
